documents/extension/funding/service.go

```diff
@@ -264,4 +274,42 @@ func (s service) DeriveFundingResponse(model documents.Model, fundingID string)
 		Header: h,
 		Data:   data,
 	}, nil
+
+}
+
+// DeriveFundingListResponse returns a funding list in client format
+func (s service) DeriveFundingListResponse(model documents.Model) (*clientfundingpb.FundingListResponse, error) {
+	response := new(clientfundingpb.FundingListResponse)
+
+	h, err := documents.DeriveResponseHeader(s.tokenRegFinder(), model)
+	if err != nil {
+		return nil, errors.New("failed to derive response: %v", err)
+	}
+	response.Header = h
+
+	fl, err := documents.AttrKeyFromLabel(fundingLabel)
+	if err != nil {
+		return nil, err
+	}
+
+	if model.AttributeExists(fl) {
+		lastIdx, err := getFundingsLatestIDX(model)
+		if err != nil {
+			return nil, err
+		}
+
+		i, err := documents.NewInt256("0")
+		if err != nil {
+			return nil, err
+		}
+
+		for ; i.Cmp(lastIdx) != 1; i.Inc() {
+			funding, err := s.deriveFundingData(model, i.String())
+			if err != nil {
+				continue
```

Comment:
Why would it fail derive here for a known index?

Reply:
Currently, there is no real case. 

I don't know if we implement `Delete` funding agreement in the future. 

If we do it, it might be simpler to just remove the index instead of reordering the arrays in the map.  In that case, it could be that an index is not existing and we `continue`

We could also throw an error for now.